SLAMNode[] nodes
VisionFactor[] vision_factors
OdometryFactor[] odometry_factors